R/HTMLBlock.R

```diff
@@ -0,0 +1,46 @@
+#' @title `HTMLBlock`
+#' @docType class
+#' @description
+#' Specialized `FileBlock` for managing HTML content in reports.
+#' It's designed to handle various HTML content, and render the report as HTML,
+#' however `htmlwidgets` objects can also be rendered to static document-ready format.
+#'
+#' @keywords internal
+HTMLBlock <- R6::R6Class( # nolint: object_name_linter.
+  classname = "HTMLBlock",
+  inherit = FileBlock,
+  public = list(
+    #' @description Initialize a `HTMLBlock` object.
+    #'
+    #' @param content An object that can be rendered as a HTML content assigned to
+    #'   this `HTMLBlock`
+    #'
+    #' @return Object of class `HTMLBlock`, invisibly.
+    initialize = function(content) {
+      if (!missing(content)) {
```

Comment:
```suggestion
      if (!missing(content)) {
        checkmate::assert_multi_class(content, c("shiny.tag", "shiny.tag.list"))
```

Reply:
Thanks, I will also include `htmlwidget` here. That is the main object type we want to support because it can be rendered in all the report types (even static) because the `print` method for the `htmlwidget` object creates the HTML content from the widget object.

```suggestion
      if (!missing(content)) {
        checkmate::assert_multi_class(content, c("shiny.tag", "shiny.tag.list", "htmlwidget"))
```